certbot-apache/certbot_apache/tests/debian_test.py

```diff
@@ -45,10 +45,11 @@ def test_enable_mod_unsupported_dirs(self):
         self.assertRaises(
             errors.NotSupportedError, self.config.enable_mod, "ssl")
 
+    @mock.patch("certbot.reverter._write_json_file")
     @mock.patch("certbot.util.run_script")
     @mock.patch("certbot.util.exe_exists")
     @mock.patch("certbot_apache.parser.subprocess.Popen")
-    def test_enable_mod(self, mock_popen, mock_exe_exists, mock_run_script):
+    def test_enable_mod(self, mock_popen, mock_exe_exists, mock_run_script, _):
```

Comment:
It was due to a hidden flaw in this kind of unit tests in fact. Normally, the reverter module will write the revert command of a command executed in the relevant storage file. So this an array of strings, basically. However, since `mock_run_scripts` is mocked here, what you get from it as a revert command is not an array of strings, it is a MagicMock  instance.

With the implementation where you just write this down directly in a file, it does not raise an exception, because you can always get a string representation of every object, even if its the default one (something like `"<MagicMock id='2373950194576'>"`).

However, when the storage was a json file, then it would throw an exception because MagicMock instances are not serializable, on the contrary to basic types (str, int) and basic containers (dict, list).

All of this to said that with the pure text representation, it is not a problem anymore, and I remove this code. But it wanted to write down somewhere that theses kind of unit tests will have funny behaviors if you try to invoke any logic about fallback in them.